cmd/agent/subcommands/run/command.go

```diff
@@ -445,6 +447,9 @@ func getSharedFxOption() fx.Option {
 		rdnsquerierfx.Module(),
 		snmptraps.Bundle(),
 		collectorimpl.Module(),
+		fx.Provide(func(_ config.Component, demux demultiplexer.Component) (ddgostatsd.ClientInterface, error) {
+			return aggregator.NewStatsdDirect(demux)
+		}),
```

Comment:
We could have the demultiplexer comp provide the conversion function directly rather than adding it to all mains.

is adding this function [here](https://github.com/DataDog/datadog-agent/blob/db20957489482a87d01501ad8677c93783950788/comp/aggregator/demultiplexer/demultiplexerimpl/demultiplexer.go#L31-L36)

Side question: why do we need the config as a parameter ?

Reply:
This is the only place the function `aggregator.NewStatsdDirect` is used. In tests cases, it is the no-op client, and within the process-agent it is the normal statsd client.

Reply:
> why do we need the config as a parameter ?

We don't, removed.